src/lib/pages/evm-tx-details/components/evm-event-box/index.tsx

```diff
@@ -0,0 +1,160 @@
+import { Flex, Stack, Text } from "@chakra-ui/react";
+import { Fragment, useState } from "react";
+import { ExplorerLink } from "lib/components/ExplorerLink";
+import { CustomIcon } from "lib/components/icon";
+import { LabelText } from "lib/components/LabelText";
+import { Tooltip } from "lib/components/Tooltip";
+import { TypeSwitch } from "lib/components/TypeSwitch";
+import type { TxReceiptJsonRpcLog } from "lib/services/types";
+import type { EvmVerifyInfo, Nullable } from "lib/types";
+import { evmParseLog } from "lib/utils";
+import { EvmEventBoxData } from "./evm-event-box-data";
+import { EvmEventBoxTopics } from "./evm-event-box-topics";
+import { EvmEventBoxTabs } from "../../types";
+
+interface EvmEventBoxProps {
+  log: TxReceiptJsonRpcLog;
+  evmVerifyInfo: Nullable<EvmVerifyInfo>;
+}
+
+export const EvmEventBox = ({ log, evmVerifyInfo }: EvmEventBoxProps) => {
+  const [currentTab, setCurrentTab] = useState(EvmEventBoxTabs.Hex);
+  const parsedLog = evmParseLog(evmVerifyInfo?.abi ?? [], log);
+
+  const handleGenerateLogName = () => {
+    if (!parsedLog) return "";
+    const { fragment } = parsedLog;
+    return (
+      <>
+        {fragment.name}(
+        {fragment.inputs.map((input, index) => (
+          <Fragment key={input.name}>
+            <Text as="span" color="success.main">
+              {input.type}
+            </Text>{" "}
+            <Text as="span" color="text.dark">
+              {input.indexed ? "indexed " : ""}
+            </Text>
+            <Text as="span" color="warning.main">
+              {input.name}
+            </Text>
+            {index < fragment.inputs.length - 1 ? ", " : ""}
+          </Fragment>
+        ))}
+        )
+      </>
+    );
+  };
```

Comment:
It's better to either move this function out of EvmEventBox component to avoid recreating it on each render or wrap it with useMemo.